components/sync_manager/android/src/main/java/mozilla/appservices/syncmanager/Errors.kt

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.appservices.syncmanager
+
+/**
+ * Base class for sync manager errors. Generally should not
+ * have concrete instances.
+ */
+open class SyncManagerException(msg: String) : Exception(msg)
```

Comment:
Maybe [sealed classes](https://phauer.com/2019/sealed-classes-exceptions-kotlin/) could be an alternative here.